sdk/batch/azure-batch/azure/batch/_operations/_patch.py

```diff
@@ -0,0 +1,776 @@
+        :keyword etag: check if resource is changed. Set None to skip checking etag. Default value is
+         None.
+        :paramtype etag: str
+        :keyword match_condition: The match condition to use upon the etag. Default value is None.
+        :paramtype match_condition: ~azure.core.MatchConditions
+        :return: None
+        :rtype: None
+        :raises ~azure.core.exceptions.HttpResponseError:
+
+        Example:
+            .. code-block:: python
+
+                # JSON input template you can fill out and use as your body input.
+                content = {
+                    "autoScaleEvaluationInterval": "1 day, 0:00:00",  # Optional. The time
+                      interval at which to automatically adjust the Pool size according to the
+                      autoscale formula. The default value is 15 minutes. The minimum and maximum value
+                      are 5 minutes and 168 hours respectively. If you specify a value less than 5
+                      minutes or greater than 168 hours, the Batch service rejects the request with an
+                      invalid property value error; if you are calling the REST API directly, the HTTP
+                      status code is 400 (Bad Request). If you specify a new interval, then the
+                      existing autoscale evaluation schedule will be stopped and a new autoscale
+                      evaluation schedule will be started, with its starting time being the time when
+                      this request was issued.
+                    "autoScaleFormula": "str"  # Optional. The formula for the desired number of
+                      Compute Nodes in the Pool. The formula is checked for validity before it is
+                      applied to the Pool. If the formula is not valid, the Batch service rejects the
+                      request with detailed error information. For more information about specifying
+                      this formula, see Automatically scale Compute Nodes in an Azure Batch Pool
+                      (https://azure.microsoft.com/en-us/documentation/articles/batch-automatic-scaling).
+                }
+        """
+        content = _models.BatchPoolEnableAutoScaleContent(
+            auto_scale_formula=auto_scale_formula,
+            auto_scale_evaluation_interval=auto_scale_evaluation_interval
+        )
+        args = [pool_id, content]
+        kwargs.update(
+            {
+                "time_out_in_seconds": time_out_in_seconds,
+                "ocpdate": ocpdate,
+                "if_modified_since": if_modified_since,
+                "if_unmodified_since": if_unmodified_since,
+                "etag": etag,
+                "match_condition": match_condition,
+            }
+        )
+        return super().enable_pool_auto_scale(*args, **kwargs)
+    
+    def terminate_job(  # pylint: disable=inconsistent-return-statements
+        self,
+        job_id: str,
+        reason: Optional[str] = None,
```

Comment:
should `reason` be a positional parameter or can we move to keyword-only?

Reply:
I believe `reason` should remain as a positional parameter in this case.